python/common/org/python/types/Str.java

```diff
@@ -708,10 +708,21 @@ private long toPositiveIndex(long index) {
     }
 
     @org.python.Method(
-            __doc__ = ""
+            __doc__ = "string.isalnum() -> bool\n" +
+                    "\n" +
+                    "Returns false when the string has atleast ONE character which is\n" +
+                    "not an alphanumeric or if the string is empty.\n"
     )
     public org.python.Object isalnum() {
-        throw new org.python.exceptions.NotImplementedError("isalnum() has not been implemented.");
+        if (!this.value.isEmpty()) {
```

Comment:
For both methods, can you shift the logic flow such that if it's empty, it returns false first? This will let you 'flatten' the rest of the logic so that it's clearer what the method is doing. Right now there are these three return statements and you have to kind of read it very carefully to see what's going on.

i.e.
```
if (empty) {
    return false
}
for (...) {
    if (...) {
        return false
    }
}
return true
```

(Some of the other methods have these huge if/elseif/elseif blocks that more or less encompass the whole method, but that's usually clearer when we need to handle different parameter types separately.)